### PR TITLE
Clean up AndroidHandleManagerTests

### PR DESCRIPTION
**Pull request by jblebrun**
Continue taking advantage of the @cromwellian pattern

### CODE REVIEW

File: javatests/arcs/android/storage/handle/AndroidHandleManagerTestBase.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import androidx.test.core.app.ApplicationProvider
+import androidx.work.testing.WorkManagerTestInitHelper
+import arcs.core.storage.handle.HandleManagerTestBase
+import com.nhaarman.mockitokotlin2.mock
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+
+@Suppress("EXPERIMENTAL_API_USAGE")
+open class AndroidHandleManagerTestBase : HandleManagerTestBase(), LifecycleOwner {
```
[the remark this answers is not in the document]
Comment (jasonwyatt) [reply javatests/arcs/android/storage/handle/AndroidHandleManagerTestBase.kt]: I think DRY is not necessarily something that needs to apply to Tests.

[the remark this answers is not in the document]
Comment (jasonwyatt) [reply javatests/arcs/android/storage/handle/AndroidHandleManagerTestBase.kt]: Some context: https://ncorti.com/blog/junit-rules

[the remark this answers is not in the document]
Comment (jblebrun) [reply javatests/arcs/android/storage/handle/AndroidHandleManagerTestBase.kt]: Is there a way to run the same set of tests multiple times with different @TestRules?

[the remark this answers is not in the document]
Comment (jblebrun) [reply javatests/arcs/android/storage/handle/AndroidHandleManagerTestBase.kt]: I'm somewhat familiar with test rules. In all of the examples I see, they seem to provide a better pattern than creating subclasses with setup.

The pattern here is the other way around: the *tests* are in the base, and the subclasses are providing setup. 

I'm struggling to figure out how to use TestRules to implement this pattern.

[the remark this answers is not in the document]
Comment (jblebrun) [reply javatests/arcs/android/storage/handle/AndroidHandleManagerTestBase.kt]: (I am happy to delay this PR a bit to refine the pattern)

[the remark this answers is not in the document]
Comment (jblebrun) [reply javatests/arcs/android/storage/handle/AndroidHandleManagerTestBase.kt]: Parameterized tests (https://github.com/junit-team/junit4/wiki/Parameterized-tests) seem to be the right way to do this. Unfortunately, `Parameterized` is a test runner, so there's currently no way to run parameterized tests and also use the `AndroidJUnit4` runner.

[the remark this answers is not in the document]
Comment (jblebrun) [reply javatests/arcs/android/storage/handle/AndroidHandleManagerTestBase.kt]: (I will apply your @TestRule request to the common setup patterns, though)

**Pull request merged**